lib/phoenix/controller.ex

```diff
@@ -834,6 +834,19 @@ defmodule Phoenix.Controller do
     put_private(conn, :phoenix_router_url, url)
   end
 
+  @doc """
+  Puts the URL or `%URI{}` to be used for the static url generation.
+
+  This function overrides the default URL generation pulled
+  from the `%Plug.Conn{}`'s endpoint configuration.
```

Comment:
Let's mention in the docs what you said in the issues description: that this is the same as `put_router_url` but it is for static urls. :)

Reply:
@josevalim I just updated the PR to include a reference to the `put_router_url` function (also applied some rewording to hopefully describe a bit clearer what the function does).